### PR TITLE
[SM-6253] Dynamic schema migrations for postgres plugin

### PR DESCRIPTION
**Pull request by jmull3n**

### CODE REVIEW

**Review by jhaynie — CHANGES_REQUESTED** [review]
mostly style nits but looking good...

does make me think we should continue to genralize this and open source so we could share between eds and wrench

File: internal/provider/postgresql.go [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+		}
+
+		if shouldCreate {
+			for i, field := range m.Fields {
+				// check if field is in payload
+				if _, ok := data[field.Name]; !ok {
+					continue
+				}
+				// if yes, then add column
+				sqlColumns.WriteString(fmt.Sprintf(`"%s"`, field.Name))
```
[the remark this answers is not in the document]
Comment (jmull3n) [reply internal/provider/postgresql.go:167]: I tried parameterizing the column names, but was getting some weird errors.

File: internal/provider/postgresql.go [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+				sqlValues.WriteString(fmt.Sprintf(`$%d`, columnCount))
+				values = append(values, data[field.Name])
+
+				if i+1 < len(m.Fields) {
+					sqlColumns.WriteString(",")
+					sqlValues.WriteString(",")
+				}
+				columnCount += 1
+			}
+			sql.WriteString(fmt.Sprintf(`INSERT INTO "%s" (%s) VALUES (%s) ON CONFLICT DO NOTHING`, m.Table, sqlColumns.String(), sqlValues.String()) + ";\n")
```
[the remark this answers is not in the document]
Comment (jmull3n) [reply internal/provider/postgresql.go]: These are just the value placeholders, so renamed the var to reflect

File: internal/provider/postgresql.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmull3n) [reply internal/provider/postgresql.go:165]: Is it bad to generate this SQL?

File: internal/provider/postgresql.go [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+	}, nil
+}
+
+// Start the provider and return an error or nil if ok
+func (p *PostgresProvider) Start() error {
+	p.logger.Info("start")
+
+	db, err := pgxpool.New(p.ctx, p.url)
+	if err != nil {
+		p.logger.Error("unable to create connection pool: %w", err)
```
Comment (jhaynie) [inline internal/provider/postgresql.go:51]: fix this in next PR but need to return this and not swallow

File: internal/provider/postgresql.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhaynie) [reply internal/provider/postgresql.go:165]: this looks good yes

**Review by jhaynie — APPROVED** [review]
a couple of comments but looks great!

**Pull request merged**